kotlin/src/test/kotlin/io/smallrye/mutiny/coroutines/MultiAsFlowTest.kt

```diff
@@ -0,0 +1,295 @@
+package io.smallrye.mutiny.coroutines
+
+import io.smallrye.mutiny.Multi
+import io.smallrye.mutiny.subscription.MultiEmitter
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.async
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.buffer
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.assertj.core.api.Assertions.assertThat
+import java.time.Duration
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicReference
+import kotlin.system.measureTimeMillis
+import kotlin.test.Test
+import kotlin.test.assertFails
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+@ExperimentalCoroutinesApi
+class MultiAsFlowTest {
```

Comment:
There are 2 dropped exceptions (on cancellation) being logged by Mutiny.

@heubeck While the tests pass I just want to double-check with you that this is fine.

Reply:
@jponge 
In my understanding the dropped exceptions are because of races between the subscription cancellation and the continues item emission.

While the coroutine is already cancelled there are still items emitted from the Multi until the subscription cancellation was propagated - what is also done as soon as `onItem` is hit by the dropped exception right?

I assume the behaviour is correct, but if the dropped exceptions should be avoided at all, I can add a `if (parentCtx.isActive)` around the `runBlocking` block - had this in before but thought it would be more nice to see (at leased in the logs and dropped exception list) that there were these race.

Reply:
btw: the same could happen in `Uni.awaitSuspending` with `continuation.resume(item)` but there it is less likely.
If we decide for the coroutine `isActive` check, I would add it there too.

Reply:
Dropped exceptions are exceptions we report somewhere (e.g., logs) because there is no other way to propagate them (e.g., the downstream has already been cancelled due to an error or completion).

So there are cases where there is really nothing we can do, but in general we shall always investigate why these exceptions happen in the first place.

Under reactive streams semantics you can have pending events being emitted after a cancellation, but here it looks like events are still being sent to an already completed downstream, so a check would probably avoid that indeed.

Can you experiment and see if we keep a correct behaviour yet avoid these dropped exceptions? Thanks!